src/uct/sm/mm/posix/mm_posix.c

```diff
@@ -399,15 +406,43 @@ uct_posix_mem_attach_common(uct_mm_seg_id_t seg_id, size_t length,
     return status;
 }
 
+static inline int uct_posix_is_same_ns(uct_mm_seg_id_t seg_id,
+                                       ucs_sys_namespace_type_t type,
+                                       const uct_posix_device_addr_t *addr)
+{
+    ucs_sys_ns_t value;
+
+    if (seg_id & UCT_POSIX_SEG_FLAG_NS) {
+        if (type == UCS_SYS_NS_TYPE_PID) {
+            value = addr->pid_ns;
+        } else {
+            ucs_assert(type == UCS_SYS_NS_TYPE_USER);
```

Comment:
minor: better to use `assertv` and print the type

Reply:
added "type=%d"